Assets/LeapMotion/Core/Scripts/Attributes/Editor/CombinablePropertyDrawer.cs

```diff
@@ -36,24 +42,35 @@ public class CombinablePropertyDrawer : PropertyDrawer {
                              property.propertyType + ".");
               continue;
             }
-            attributes.Add(combinableProperty);
+            outAttributes.Add(combinableProperty);
           }
         }
 
-        _cachedAttributes[fieldInfo] = attributes;
+        _cachedAttributes[fieldInfo] = outAttributes;
       }
     }
 
-    public override void OnGUI(Rect position, SerializedProperty property, GUIContent label) {
+    public override void OnGUI(Rect position, SerializedProperty property,
+                               GUIContent label) {
       getAttributes(property);
 
+      CombinablePropertyDrawer.OnGUI(this.attributes, this.fieldInfo,
+                                     position, property, label);
+    }
+
+    public static void OnGUI(List<CombinablePropertyAttribute> attributes,
```

Comment:
Out of curiosity, what kind of work are you doing that requires calling into the OnGUI code of CombinablePropertyDrawer directly?

Reply:
will investigate :D there waaaas almost certainly a good reason

Reply:
Here's where I used it.
 
   private void drawStreamProperty(SerializedProperty property) {
      var rect = EditorGUILayout.GetControlRect(true);
      var label = new GUIContent("Stream ("
                                 + (target as StreamConnector).streamDataType.Name + ")");

      var streamType = (target as StreamConnector).streamType;

      var spoofedImplementsInterfaceAttribute
        = new ImplementsInterfaceAttribute(streamType);

      var attributes = Pool<List<CombinablePropertyAttribute>>.Spawn();
      attributes.Clear();
      try {
        attributes.Add(spoofedImplementsInterfaceAttribute);
        CombinablePropertyDrawer.OnGUI(attributes, null, rect, property, label);
      }
      finally {
        attributes.Clear();
        Pool<List<CombinablePropertyAttribute>>.Recycle(attributes);
      }
    }

Reply:
Whole file here:
https://github.com/leapmotion/LeapPaint/blob/master/Assets/AppModules/AssetTools/Streams/Editor/StreamConnectorEditor.cs